### PR TITLE
🚀 Execute Async Functions Concurrently with `Promise.all`

### PR DESCRIPTION
**Pull request by sanjaiyan-dev**
This PR implements concurrent execution of async functions using `Promise.all`. The advantages of this approach include:

- Improved performance
- Reduced execution time
- Optimized resource utilization
- Enhanced scalability
- Avoidance of blocking operations
- Simplified error handling
- Clearer code structure
- Consistent API design

Please review and provide feedback.

Thank you!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,9 +10,9 @@
...
 
 if (!Array.prototype.forEachAsync) {
-    Array.prototype.forEachAsync = async function(callback) {
-        for (let index = 0; index < this.length; index++) {
-            await callback(this[index], index, this);
-        }
+    Array.prototype.forEachAsync = async function (callback) {
+        await Promise.all(this.map((item, index) => {
+            return callback(item, index, this);
+        }));
```
Comment (mingchen) [inline index.js]: This modification shifts the behavior from sequential execution to parallel execution, potentially leading to unforeseen outcomes. It is recommended to introduce a new parameter to accommodate and facilitate this functionality. e.g. 
```
Array.prototype.forEachAsync = async function (callback, parallel = false) {
  if (parallel) {
      // new logical
  } else {
     // old logical
  }
}
```
comments should also be updated.

Comment (sanjaiyan-dev) [reply index.js]: > This modification shifts the behavior from sequential execution to parallel execution, potentially leading to unforeseen outcomes. It is recommended to introduce a new parameter to accommodate and facilitate this functionality. e.g.
> 
> ```
> Array.prototype.forEachAsync = async function (callback, parallel = false) {
>   if (parallel) {
>       // new logical
>   } else {
>      // old logical
>   }
> }
> ```
> 
> comments should also be updated.

Done :)

**Review by mingchen — APPROVED** [review]
Thanks for contribution and changes.

**Comment by sanjaiyan-dev** [comment]
> 

Thank you 🙏🏽

**Comment by sanjaiyan-dev** [comment]
> Thanks for contribution and changes.

Thank you 🙏🏽

**Pull request merged**